src/middlewared/middlewared/plugins/replication.py

```diff
@@ -431,26 +270,15 @@ async def do_update(self, app, audit_callback, id_, data):
 
         return await self.get_instance(id_)
 
-    @accepts(
-        Int("id"),
+    @api_method(
+        ReplicationDeleteArgs,
+        ReplicationDeleteResult,
         audit="Replication task delete:",
-        audit_callback=True
+        audit_callback=True,
```

Comment:
Need to add `roles=` to the crud endpoints while you're here.

Reply:
Why? They're added automatically.

Reply:
Ahh, right sorry. I forgot about the prefix attribute in the nested config attribute of this class.